### DISCUSSION TITLE
Tiny objects can be degenerated to spheres

### DISCUSSION THREAD

**Discussion by labbbirder** [notgiven688/jitterphysics2#150] [Ideas]
Hi, bro @notgiven688. This is a topic from #143 . Sorry but I wanna a further discussion on this problem now.

As the conclusion from a physics point of view, I tried to limit the size from the shape authoring side. However, I found it's not a neat solution because of the variety of shapes. And it is unpredictable to find a min float to avoid arithmetic overflow, as there are different platforms with different floating-control behavior and relevant API.

Imagine that a scene with physics simulator running, those tiny shapes are not noticeable at all, as an assumption that the sizes in physical and graphical side are approximately the same.

So, I propose that when a shape is too small, it can be degenerated to a sphere with a inverse inertia tensor equals a identity matrix multiplied by a proper large number. Because the rotation acts like a sphere for tiny object is absolutely acceptable and reasonable.

How do you think about that, I'll be glad to see ur opinion.

**Comment by notgiven688** [comment]
> As the conclusion from a physics point of view, I tried to limit the size from the shape authoring side.

This should be the way to go. Calculate the bounding box and check that no side is smaller than 0.01 unit and you are fine.

Everything else is nonsense. Jitter is not made to handle smaller dynamic objects - at least not in 32bit floating point mode.

edit: added "dynamic"
edit2: body.AddShape(shape, setMassInertia: false) can always be used to ignore the shape's mass properties.
